integration-tests/reactive-messaging-rabbitmq-vault/src/main/java/io/quarkus/it/rabbitmq/PeopleProducer.java

```diff
@@ -0,0 +1,26 @@
+package io.quarkus.it.rabbitmq;
+
+import java.time.Duration;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipse.microprofile.reactive.messaging.Outgoing;
+
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+@ApplicationScoped
+public class PeopleProducer {
+    @Outgoing("people-out")
+    public Multi<Person> generatePeople() {
+        return Uni.createFrom().nullItem()
+                .onItem().delayIt().by(Duration.ofSeconds(3))
```

Comment:
that sounds like a way to avoid race conditions. It might be better to use an emitter.

Reply:
I just remembered. This delay is because we don't have any insight or notification as to when the `RabbitMQConnector` builds the RabbitMQ queue for `people-in`. So, to check we are not missing messages we need a fairly big delay to ensure it is created before we send any messages.

This was the "fix" applied to `RabbitMQConnectorTest` and has worked properly.